cpp/src/replace/nulls.cu

```diff
@@ -277,6 +291,7 @@ struct replace_nulls_scalar_kernel_forwarder {
                                            rmm::mr::device_memory_resource* mr,
                                            cudaStream_t stream = 0)
   {
+    CUDF_EXPECTS(input.type() == replacement.type(), "Data type mismatch");
```

Comment:
Should we keep this expectation when the input is not a dictionary column?

Reply:
We did keep it. It just needed to be moved here since there is no column type DICTIONARY that matches any scalar type.

Reply:
Yeah this was a brain fart.